v2/blockstore/readwrite_test.go

```diff
@@ -1,19 +1,202 @@
 package blockstore_test
 
 import (
+	"context"
+	"fmt"
 	"io"
 	"math/rand"
 	"os"
 	"testing"
 
+	"github.com/ipfs/go-blockservice"
+	ds "github.com/ipfs/go-datastore"
+	ds_sync "github.com/ipfs/go-datastore/sync"
+	bstore "github.com/ipfs/go-ipfs-blockstore"
+	chunk "github.com/ipfs/go-ipfs-chunker"
+	offline "github.com/ipfs/go-ipfs-exchange-offline"
+	"github.com/ipfs/go-merkledag"
+	"github.com/ipfs/go-unixfs/importer/balanced"
+	"github.com/ipfs/go-unixfs/importer/helpers"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 
 	"github.com/ipld/go-car/v2/blockstore"
 
 	"github.com/ipfs/go-cid"
 	"github.com/ipld/go-car/v2/internal/carv1"
 )
 
+const unixfsChunkSize uint64 = 1 << 10
+const unixfsLinksPerLevel = 1024
+
+func TestFailure(t *testing.T) {
```

Comment:
@mvdan @masih @willscott 

These test failures are important to fix. We can go over them in our sync call.